src/dlgTriggerEditor.cpp

```diff
@@ -4738,56 +4799,168 @@ void dlgTriggerEditor::slot_var_selected(QTreeWidgetItem *pItem)
+            mpActionsMainArea->pushButton_clearIcon->setEnabled( false );
+        }
+        else
+        {
+            mpActionsMainArea->lineEdit_action_icon->setText( iconFileName );
+            mpActionsMainArea->pushButton_changeIcon->setIcon( QIcon( iconFileName ) );
+            mpActionsMainArea->pushButton_clearIcon->setEnabled( true );
+        }
+        mpActionsMainArea->lineEdit_action_command_down->setText( pT->getCommandButtonDown() );
+        mpActionsMainArea->lineEdit_action_command_up->setText( pT->getCommandButtonUp() );
         mpSourceEditor->setPlainText( pT->getScript() );
         // location = 1 = location = bottom is no longer supported
         int location = pT->mLocation;
-        if( location > 0 ) location--;
+        if( location > 0 )
+        {
+            location--;
+        }
         mpActionsMainArea->comboBox_location->setCurrentIndex( location );
         mpActionsMainArea->comboBox_orientation->setCurrentIndex( pT->mOrientation );
-        QColor color = pT->getButtonColor();
-        QPalette palette;
-        palette.setColor( QPalette::Button, color );
-        mpActionsMainArea->buttonRotation->setCurrentIndex( pT->getButtonRotation() );
-        mpActionsMainArea->buttonColumns->setValue( pT->getButtonColumns() );
-        mpActionsMainArea->css->clear();
+        mpActionsMainArea->comboBox_buttonRotation->setCurrentIndex( pT->getButtonRotation() );
+        mpActionsMainArea->spinBox_buttonColumns->setValue( pT->getButtonColumns() );
         mpActionsMainArea->css->setPlainText( pT->css );
-        if( ! pT->getParent() )
+
+        // Now show/hide the relevant bits:
+        mpActionsMainArea->show();
+        mpSystemMessageArea->hide();
+        if( pT->isFolder() && ! pT->mPackageName.isEmpty() )
         {
+            // This is a module master or package folder - hide everything but
+            // the name and protect that from editing
+            mpActionsMainArea->widget_middle->hide();
+            mpActionsMainArea->widget_bottom->hide();
+            mpSourceEditorArea->hide();
+            mpActionsMainArea->lineEdit_action_name->setReadOnly(true);
+            mpActionsMainArea->lineEdit_action_name->setToolTip( QStringLiteral( "<html><head/><body><p>%1</p></body></html>" )
+                                                                 .arg( "The module or package name was set when the module was created and cannot be modified "
+                                                                       "here. You may add or remove things from the module by dragging and dropping and all the "
+                                                                       "button/menu/toolbar items can be enabled/disabled at once by activating/deactivating "
+                                                                       "this item." ) );
+
+            mpDeleteItemButton->setEnabled( false ); // Prevent deletion on this type of action
```

Comment:
For all the tooltips you've added, you missed an important one here explaing why is the button disabled. That's not going to be always immediately obvious :)

How about:

> This is the top-level folder of a module/package - to remove it, uninstall the module/package.